xios_examples/gen_netcdf.py

```diff
@@ -0,0 +1,290 @@
+
+    face_face = ncmeshout.createVariable(f"{meshout_varname}_face_links", np.int32, (face_dim,vertex_dim), fill_value=999999)
+    face_face.cf_role = "face_face_connectivity"
+    face_face.long_name = "Indicates which other faces neighbor each face"
+    face_face.start_index = np.int32(start_index)
+    face_face.flag_values = np.int32(-1) ;
+    face_face.flag_meanings = "out_of_mesh" ;
+    face_face[:] = face_face_connectivity[:] - face_face_connectivity.start_index + start_index
+
+    if add_bounds:
+        face_x.bounds = f"{face_x.name}_bounds"
+        face_x_bnds = ncmeshout.createVariable(face_x.bounds, face_x.dtype, face_node.dimensions)
+        face_x_bnds[:] = node_x[face_node[:].flatten()].reshape(face_x_bnds.shape)
+        
+        face_y.bounds = f"{face_y.name}_bounds"
+        face_y_bnds = ncmeshout.createVariable(face_y.bounds, face_y.dtype, face_node.dimensions)
+        face_y_bnds[:] = node_y[face_node[:].flatten()].reshape(face_y_bnds.shape)
+
+    data = ncmeshout.createVariable(dataname, np.float64, (face_dim,))
+    data.long_name = "input data values"
+    data.mesh = meshout_varname
+    data.location = "face"
+    data.coordinates = f"{face_y.name} {face_x.name}"
+    data[:] = func(face_lat, face_lon)
+
+    ncmeshin.close()
+
+def getargs(argv=None):
+
+    df = dataFunc()
+    funclist = df.get_funclist()
+    del df
+
+    parser = argparse.ArgumentParser(description="Generate netCDF files with data on domains suitable for regridding")
+
+    parser.add_argument("--meshfile", help="Name of netCDF file containing UGRID mesh topology data, needed for UGRID data", dest='mesh_file')
+    parser.add_argument("--meshvar", help="Variable name of mesh topology data in netCDF file, optional for UGRID data", dest='mesh_varname')
+    parser.add_argument("--func", help="Analytic function for data variable (default: %(default)s)", choices=funclist, dest='func_str')
+    parser.add_argument("--nlat", help="Number of latitude points for original grid, not needed for UGRID data (default: %(default)d)", type = int)
+    parser.add_argument("--nlon", help="Number of longitude points for original grid, not needed for UGRID data (default: %(default)d)", type = int)
+    parser.add_argument("--nlatr", help="Number of latitude points for resampled grid (default: %(default)d)", type = int)
+    parser.add_argument("--nlonr", help="Number of longitude points for resampled grid (default: %(default)d)", type = int)
+    parser.add_argument("file_out", help="Name of netCDF non-UGRID output file")
+
+    parser.set_defaults(**defaults)
+    args = parser.parse_args(argv)
+
+    return args
+
+def run(file_out, func_str=defaults['func_str'], mesh_file=defaults['mesh_file'], 
+                  mesh_varname=defaults['mesh_varname'], 
+                  nlat=defaults['nlat'], nlon=defaults['nlon'], 
+                  nlatr=defaults['nlatr'], nlonr=defaults['nlonr']):
+
```

Comment:
a docstring would help here.
One unknown on reading: can file_out be a path that includes directories and file name, or is it limited to file name within current working dir? (see shared_testing comments)

Reply:
Docstring added. Yes you can use full paths.